src/H5Tref.c

```diff
@@ -980,12 +979,14 @@ H5T__ref_disk_write(H5VL_object_t H5_ATTR_UNUSED *src_file, const void *src_buf,
     /* TODO Should get rid of bg stuff */
     if (p_bg) {
         H5VL_blob_specific_args_t vol_cb_args; /* Arguments to VOL callback */
-        size_t                    p_buf_size_left = dst_size;
 
         /* Skip the size / header */
         p_bg += (sizeof(uint32_t) + H5R_ENCODE_HEADER_SIZE);
+
+#ifndef NDEBUG
+        size_t p_buf_size_left = dst_size;
         HDassert(p_buf_size_left > (sizeof(uint32_t) + H5R_ENCODE_HEADER_SIZE));
-        p_buf_size_left -= (sizeof(uint32_t) + H5R_ENCODE_HEADER_SIZE);
+#endif
```

Comment:
The changes in this function just isolate the sanity checks to NDEBUG blocks to avoid 'unused' warnings in production mode